java/aiservices/openai/src/main/java/com/microsoft/semantickernel/aiservices/openai/chatcompletion/OpenAIChatCompletion.java

```diff
@@ -76,46 +72,81 @@ public Map<String, ContextVariable<?>> getAttributes() {
 
         List<ChatRequestMessage> chatRequestMessages = getChatRequestMessages(chatHistory);
         List<FunctionDefinition> functions = Collections.emptyList();
-        return internalChatMessageContentsAsync(chatRequestMessages, functions, promptExecutionSettings);
+        return internalChatMessageContentsAsync(chatRequestMessages, functions,
+            promptExecutionSettings);
 
     }
 
     @Override
-    public Flux<StreamingChatMessageContent> getStreamingChatMessageContentsAsync(ChatHistory chatHistory,
-            PromptExecutionSettings promptExecutionSettings, Kernel kernel)
-    {
-        List<ChatRequestMessage> chatRequestMessages = getChatRequestMessages(chatHistory);
-        List<FunctionDefinition> functions = Collections.emptyList();
-        return internalStreamingChatMessageContentsAsync(chatRequestMessages, functions, promptExecutionSettings);
-
+    public Flux<StreamingChatMessageContent> getStreamingChatMessageContentsAsync(
+        ChatHistory chatHistory,
+        PromptExecutionSettings promptExecutionSettings, Kernel kernel) {
+
+        return internalStreamingChatMessageContentsAsync(
+            getChatRequestMessages(chatHistory),
+            Collections.emptyList(),
+            promptExecutionSettings);
     }
 
     @Override
     public Mono<List<ChatMessageContent>> getChatMessageContentsAsync(String prompt,
-            PromptExecutionSettings promptExecutionSettings, Kernel kernel) {
+        PromptExecutionSettings promptExecutionSettings, Kernel kernel) {
         List<ChatRequestMessage> chatRequestMessages = getChatRequestMessages(prompt);
         List<FunctionDefinition> functions = getFunctionDefinitions(prompt);
-        return internalChatMessageContentsAsync(chatRequestMessages, functions, promptExecutionSettings);
+        return internalChatMessageContentsAsync(chatRequestMessages, functions,
+            promptExecutionSettings);
     }
 
     @Override
-    public Flux<StreamingChatMessageContent> getStreamingChatMessageContentsAsync(String prompt,
-            PromptExecutionSettings promptExecutionSettings, Kernel kernel) {
-        List<ChatRequestMessage> chatRequestMessages = getChatRequestMessages(prompt);
-        List<FunctionDefinition> functions = getFunctionDefinitions(prompt);
-        return internalStreamingChatMessageContentsAsync(chatRequestMessages, functions, promptExecutionSettings);
+    public Flux<StreamingChatMessageContent> getStreamingChatMessageContentsAsync(
+        String prompt,
+        PromptExecutionSettings promptExecutionSettings,
+        Kernel kernel) {
+
+        List<String> prompts = Arrays.asList(
+            prompt,
+            "<prompt>" + prompt + "</prompt>"
+        );
+
+        List<ChatRequestMessage> messages = Collections.singletonList(
+            new ChatRequestUserMessage(prompt));
+
+        List<FunctionDefinition> functions = null;
+
+        for (String p : prompts) {
```

Comment:
This needs pushed into each of getChatMessages and getFunctionDefinitions since those are called from multiple places. 

Really, there should be a "parsePrompt" that does this and returns a List<XMLEvent> which can then be used to extract messages and functions. 

See also the comment I left in getChatRequestMessages.